plone/app/users/browser/register.py

```diff
@@ -60,9 +72,9 @@ class BaseRegistrationForm(AutoExtensibleForm, form.Form):
     # this attribute indicates if user was successfully registered
     _finishedRegister = False
 
-    def __init__(self, *args, **kwargs):
-        super(BaseRegistrationForm, self).__init__(*args, **kwargs)
-        self.schema = getUtility(IRegisterSchemaProvider).getSchema()
+    @property
```

Comment:
I understand your wish to have the code run only once. I had to disable caching as it was broken. I think that the next step is to enable proper caching and ensure that code is run once.